services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -804,14 +806,14 @@ testNoConsentBlockOne2OneConv connectFirst teamPeer approveLH testPendingConnect
         liftIO $ assertEqual "approving should change status" (if approveLH then UserLegalHoldEnabled else UserLegalHoldPending) userStatus
         if approveLH
           then
-            getInternalClientsFull (UserSet $ Set.fromList [legalholder])
-              <&> userClientsFull
-              <&> Map.elems
-              <&> Set.unions
-              <&> Set.toList
-              <&> (\[x] -> x)
-              <&> clientId
-              <&> Just
+            getInternalClientsFull (UserSet $ Set.singleton legalholder)
+              <&> do
+                userClientsFull
+                  >>> Map.elems
+                  >>> Set.unions
+                  >>> Set.toList
+                  >>> listToMaybe
+                  >>> fmap clientId
```

Comment:
What do we get by doing these changes?

Reply:
- hlint is happy
- mapping only once (not reconstructing the Set if GHC doesn't optimize)

Reply:
also if you mean the `s/\[x] -> x/listToMaybe` change, I think it's nice to avoid partial pattern matches where possible.